authorization/storage_authorization.go

```diff
@@ -50,6 +81,46 @@ func decodeAuthorization(b []byte, a *influxdb.Authorization) error {
 	return nil
 }
 
+// transformToken updates a.Token and a.HashedToken to match configuration state,
+// if needed. If needed, transformToken generates the a.HashedToken from a.Token when
+// token hashing is enabled. transformToken will also clear a.HashedToken if token
+// hashing is turned off and a.Token is set to the matching token. If a.HashedToken and
+// a.Token are both set but do not match (a.HashedToken is a hash of a.Token), then an
+// error is returned.
+func (s *Store) transformToken(a *influxdb.Authorization) error {
+	// Verify Token and HashedToken match if both are set.
+	if a.Token != "" && a.HashedToken != "" {
```

Comment:
since 'a' is a pointer should there be a nil check for safety before field access?

Reply:
Shouldn't happen, but wouldn't hurt to add it. Will be in next commit.

Reply:
I decided to add the nil check in the public methods that act as gateways to these methods.